pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -710,9 +744,15 @@ func compareCAPIInfraMachineTemplates(platform configv1.PlatformType, infraMachi
 			return nil, errAssertingCAPIIBMPowerVSMachineTemplate
 		}
 
-		var diff []string
-		diff = append(diff, deep.Equal(typedInfraMachineTemplate1.Spec, typedinfraMachineTemplate2.Spec)...)
-		diff = append(diff, util.ObjectMetaEqual(typedInfraMachineTemplate1.ObjectMeta, typedinfraMachineTemplate2.ObjectMeta)...)
+		diff := make(map[string]any)
+
+		if diffSpec := deep.Equal(typedInfraMachineTemplate1.Spec, typedinfraMachineTemplate2.Spec); len(diffSpec) > 0 {
+			diff[".spec"] = diffSpec
+		}
+
+		if diffObjectMeta := deep.Equal(typedInfraMachineTemplate1.ObjectMeta, typedinfraMachineTemplate2.ObjectMeta); len(diffObjectMeta) > 0 {
+			diff[".metadata"] = diffObjectMeta
```

Comment:
how does this print? do we `%+v` the keys? I found the diff confusing when I was trying to debug some tests.

Reply:
Kinda

```
%v	the value in a default format
	when printing structs, the plus flag (%+v) adds field names
```, we print field names. also with the new map[string] approach we can prefix the field name when we break down the diff checks like in this case with .metadata.